File: spring-boot-oauth2-auth-server/src/main/java/com/bittokazi/oauth2/auth/server/app/controllers/open/PublicTenantController.java
package com.bittokazi.oauth2.auth.server.app.controllers.open;

import com.bittokazi.oauth2.auth.server.app.services.tenant.TenantService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/public/api")
@RequiredArgsConstructor
public class PublicTenantController {

    private final TenantService tenantService;

    @GetMapping("/tenants/info")
    public ResponseEntity<?> info() {
        return tenantService.info();
    }
}

File: spring-boot-oauth2-auth-server/src/main/kotlin/com/bittokazi/oauth2/auth/server/app/controllers/open/PublicTenantController.kt
package com.bittokazi.oauth2.auth.server.app.controllers.open

import com.bittokazi.oauth2.auth.server.app.services.tenant.TenantService
import io.swagger.v3.oas.annotations.Hidden
import lombok.RequiredArgsConstructor
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@Hidden
@RestController
@RequestMapping("/public/api")
@RequiredArgsConstructor
class PublicTenantController(
    private val tenantService: TenantService
) {

    @GetMapping("/tenants/info")
    fun info(): ResponseEntity<*> = tenantService.info()
}
